Sources/Main.cpp

```diff
@@ -1174,51 +1139,101 @@ void __fastcall TX584Form::PasteItemClick(TObject *Sender)
                 CPU.Format(Code[index + i], str);
                 CodeListView->Items->Item[index + i]->SubItems->Strings[1] = str;
                 CodeListView->Items->Item[index + i]->SubItems->Strings[2] = CMClipboard[i];
+                CodeListView->Items->Item[index + i]->Selected = true;
             }
-        SelCount = ClipboardSize;
-        if (index + SelCount > MAX_ADDR)
-            SelCount = MAX_ADDR - index;
+
         CodeListView->Repaint();
         SetModifyFlag(true);
     } else
         PostMessageW(ActiveControl->Handle, WM_PASTE, 0, 0);
 }
 //---------------------------------------------------------------------------
 
+void TX584Form::ClearSelectedItems()
+{
+    TListItem *Item;
+
+    Item = CodeListView->Selected;
+    TItemStates selected = TItemStates() << isSelected;
+    for (; Item; Item = CodeListView->GetNextItem(Item, sdBelow, selected)) {
+        int Index = Item->Index;
+        Code[Index] = NOP;
+        Item->SubItems->Strings[1] = NOP_TEXT;
+        Item->SubItems->Strings[2] = L"";
+    }
+}
+//---------------------------------------------------------------------------
+
+void TX584Form::RemoveSelectedItems()
+{
+    bool NewItems[MAX_ADDR] = {0};
```

Comment:
I'm confused about `NewItems` array. All it does is determine whether an item is selected or not. Why not simply loop through all `TListItem`s and use their `Selected` property instead of making an extra `NewItems` array? Am I missing something? :thinking: 

```cpp
int OldIndex = 0;
int NewIndex = 0;

// копируем все невыделенные инструкции в начало
for (; OldIndex < MAX_ADDR; OldIndex++) {
    if (!CodeListView->Items->Item[OldIndex]->Selected) {
        TListItem *OldItem = CodeListView->Items->Item[OldIndex];
        TListItem *NewItem = CodeListView->Items->Item[NewIndex];
        ...
    }
}
```

Reply:
I agree. NewItems array is too redundant

Reply:
Oh... I reviewed this very mindlessly...
I remember why I was needed NewItems array: I saved old Selected values into it.